src/api/general-services/work-submit/create-worker-k8s.js

```diff
@@ -1,110 +1,69 @@
-const k8s = require('@kubernetes/client-node');
+const util = require('util');
+const fs = require('fs').promises;
+const tmp = require('tmp-promise');
+const childProcess = require('child_process');
+const fetch = require('node-fetch');
+const YAML = require('yaml');
+const { Downloader } = require('github-download-directory');
 const config = require('../../../config');
 const logger = require('../../../utils/logging');
 
+const constructChartValues = async (service) => {
+  const { workQueueName } = service;
+  const { experimentId } = service.workRequest;
+  const { clusterEnv } = config;
+
+  const response = await fetch(
+    config.workerInstanceConfigUrl,
+    {
+      method: 'GET',
+    },
+  );
+
+  return response.text().then((txt) => {
+    const cfg = YAML.parse(txt);
+
+    return {
+      ...cfg,
+      experimentId,
+      clusterEnv,
+      workQueueName,
+      storageSize: '10Gi',
+    };
+  });
+};
+
 const createWorkerResources = async (service) => {
-  const namespaceName = 'worker-refs-heads-master';
+  const { workerHash } = service;
+  const HELM_BINARY = '/usr/local/bin/helm';
+  const execFile = util.promisify(childProcess.execFile);
 
-  const kc = new k8s.KubeConfig();
-  kc.loadFromDefault();
+  // Download value template from Git repository. Fill in needed things.
+  const instanceConfig = await constructChartValues(service);
+  const { name } = tmp.fileSync();
+  await fs.writeFile(name, YAML.stringify(instanceConfig));
 
-  const k8sCoreApi = kc.makeApiClient(k8s.CoreV1Api);
-  const k8sBatchApi = kc.makeApiClient(k8s.BatchV1Api);
+  // Download the chart from the worker repository.
+  const custom = new Downloader({
+    github: { auth: '64ed5ec7a15c641069525e5c43464ead8bbdc20b' },
```

Comment:
this needs to be auto generated

Reply:
i did a mix of things -- i ended up making it an org-wide secret that people have to update manually, but i can put it in the IAC repo or add a script for it.